crates/trie/common/src/proofs.rs

```diff
@@ -1,20 +1,21 @@
 //! Merkle trie proofs.
 
-use super::{traits::IntoTrieAccount, Nibbles};
-use crate::{keccak256, Account, Address, Bytes, B256, U256};
+use crate::{Nibbles, TrieAccount};
+use alloy_primitives::{keccak256, Address, Bytes, B256, U256};
 use alloy_rlp::encode_fixed_size;
 use alloy_trie::{
     proof::{verify_proof, ProofVerificationError},
     EMPTY_ROOT_HASH,
 };
+use reth_primitives_traits::Account;
 
 /// The merkle proof with the relevant account info.
 #[derive(PartialEq, Eq, Debug)]
-pub struct AccountProof {
+pub struct AccountProof<A: Account> {
```

Comment:
this is now required because we don't depend on reth-primitives/Account ?

I believe this makes sense, because we'd like to support different account objects. one use case for this would be blast for example

but I wonder if this is premature because this now leaks quite a bit